cpp/include/cudf/shifting.hpp

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef SHIFTING_HPP
+#define SHIFTING_HPP
+
+#include "cudf.h"
+
+namespace cudf {
+
+table shift(
+    table const& in_table,
+    gdf_index_type offset,
+    gdf_scalar const& fill_value
+);
+
+gdf_column shift(
```

Comment:
Since it's trivial to construct a `table` from a `gdf_column`, can we just provide the single `table` API?

Reply:
done